airflow/config_templates/config.yml

```
@@ -2121,6 +2121,14 @@
       type: integer
       example: ~
       default: "20"
+    - name: stalled_dags_update_timeout
```

I think you mean

```suggestion
    - name: deactivate_stale_dags_interval
```

This could also be left as a "hidden" option, i.e. use the `conf.getint(..., fallback=60)` you have, and just don't add it to the config.yml and default_airflow.cfg if you want to avoid config sprawl.

---

There is already config `deactivate_stale_dags_interval`:
https://github.com/apache/airflow/blob/main/airflow/config_templates/default_airflow.cfg#L982

but it is used by DagProcessor and has slightly different meaning.

I believe 60seconds there may be too low - as if dag parsing process takes more that 1 minute (many dags or high min_file_process_interval) then Scheduler may start removing some dags (as they were not updated withing last 60 seconds).
I put 10minutes there - as this sounds like a safer value.

Other idea: instead of introducing new config we may simply use `min_file_process_interval * X` with e.g X=10?

---

Oh, I see, `stalled` is something else to `stale` (stalled = not going anywhere, usually applied to a dag run or queued task instance).

Yes, using that existing one was wrong, I read the diff wrong

How about `dag_stale_not_seen_duration` `mark_dag_stale_not_seen_in` as alternative names for this config setting?

---

Renamed. Thanks!